### PR TITLE
Rename types for cross-language consistency

### LINKED ISSUE #8784: Sanitize PII samples
https://github.com/Azure/azure-sdk-for-net/pull/8677#discussion_r345381921

We should be able to use 1234 type numbers to indicate fakeness of numbers.
Take out Brazilian CFP as this is still in flux.
Reach out to service team as needed to get functional sample with fake numbers.

### LINKED ISSUE #9078: Rename TextAnalysis* to TextAnalytics*
Per JS change to this effect - otherwise it looks a bit like a typo.

Include other naming changes discussed in this as well.
https://microsoft-my.sharepoint.com/:x:/p/annelo/EVJZmHl62dRLtGmNTx0xSpYBgWn-FyAzjpCmXkEB7tPikg

### PR DESCRIPTION
**Pull request by annelo-msft**
Renaming types to be consistent with other languages, per API review from JavaScript team, and internal team discussion around common type names.

Fixes https://github.com/Azure/azure-sdk-for-net/issues/9078
Fixes https://github.com/Azure/azure-sdk-for-net/issues/8784

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizePiiEntitiesResultCollection.cs [hunk 1]
```diff
@@ -8,22 +8,22 @@ namespace Azure.AI.TextAnalytics
 {
     /// <summary>
     /// </summary>
-    public class ExtractLinkedEntitiesResultCollection : ReadOnlyCollection<ExtractLinkedEntitiesResult>
+    public class RecognizePiiEntitiesResultCollection : ReadOnlyCollection<RecognizePiiEntitiesResult>
```
Comment (tg-msft) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizePiiEntitiesResultCollection.cs]: Question for future API discussion - is it worth having a base ResultCollection<T> for these?

**Review by kristapratico — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/samples/Sample6_RecognizePiiEntities.cs [hunk 2]
```diff
@@ -22,10 +22,10 @@ public void RecognizePiiEntities()
             // Instantiate a client that will be used to call the service.
             var client = new TextAnalyticsClient(new Uri(endpoint), subscriptionKey);
 
-            string input = "A developer with SSN 859-98-0987 whose phone number is 206-867-5309 is building tools with our APIs.";
+            string input = "A developer with SSN 555-55-5555 whose phone number is 555-555-5555 is building tools with our APIs.";
```
Comment (tg-msft) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/samples/Sample6_RecognizePiiEntities.cs]: The service won't also change `Jenny` to `A developer` as part of redacting PII?  _Shame..._

**Review by tg-msft — APPROVED** [review]
Changes look good.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizePiiEntitiesResultCollection.cs (hunk 1 above)
Comment (annelo-msft) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizePiiEntitiesResultCollection.cs]: We tried this, and Krzysztof didn't like using generics for this.  But I had the same thought.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs [hunk 3]
```diff
@@ -1089,11 +1089,11 @@ public virtual Response<ExtractLinkedEntitiesResult> ExtractEntityLinking(string
         /// <param name="language"></param>
         /// <param name="cancellationToken"></param>
         /// <returns></returns>
-        public virtual async Task<Response<ExtractLinkedEntitiesResultCollection>> ExtractEntityLinkingAsync(IEnumerable<string> inputs, string language = default, CancellationToken cancellationToken = default)
+        public virtual async Task<Response<RecognizeLinkedEntitiesResultCollection>> ExtractEntityLinkingAsync(IEnumerable<string> inputs, string language = default, CancellationToken cancellationToken = default)
```
Comment (xirzec) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs]: Shouldn't this method (and related) be renamed to `RecognizeLinkedEntitiesAsync`?

Comment (annelo-msft) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs]: Yes!  Good catch - thanks!  I'll fix this.

**Pull request merged**